runtime/compiler/env/J9ClassEnv.hpp

```diff
@@ -124,16 +124,31 @@ class OMR_EXTENSIBLE ClassEnv : public OMR::ClassEnvConnector
    bool isString(TR::Compilation *comp, uintptr_t objectPointer);
    bool jitStaticsAreSame(TR::Compilation *comp, TR_ResolvedMethod * method1, int32_t cpIndex1, TR_ResolvedMethod * method2, int32_t cpIndex2);
    bool jitFieldsAreSame(TR::Compilation *comp, TR_ResolvedMethod * method1, int32_t cpIndex1, TR_ResolvedMethod * method2, int32_t cpIndex2, int32_t isStatic);
+   /*
+    * \brief
+    *    Tells whether a class reference entry in the constant pool represents a value type class.
+    *
+    * \param cpContextClass
+    *    The class whose constant pool contains the class reference entry being looked at. In another words,
+    *    it's the class of the method referring to the class reference entry.
+    *
+    * \param cpIndex
+    *    The constant pool index of the class reference entry.
+    *
+    * \note
+    *    The class reference entry doesn't need to be resolved because the information is encoded in class name string
+    */
+   bool isClassRefValueType(TR_OpaqueClassBlock *cpContextClass, int32_t cpIndex, TR::Compilation * comp);
```

Comment:
Given value types exist at the OMR level should we contribute the API to OMR and always return false in their implementation for the time being?

Reply:
I think this query is specific to constant pool entry which only exists in openj9.